# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.h
# Commit: 78181c32bdc6
# Full Hash: 78181c32bdc6b1d66eac04f3ce11a2902f477823
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r 912cf1adac80 -r 78181c32bdc6 dom/workers/WorkerScope.h
--- a/dom/workers/WorkerScope.h	Thu Feb 27 16:17:34 2025 +0000
+++ b/dom/workers/WorkerScope.h	Thu Feb 27 16:17:35 2025 +0000
@@ -86,6 +86,7 @@
 class VsyncWorkerChild;
 class WebTaskScheduler;
 class WebTaskSchedulerWorker;
+class WebTaskSchedulingState;
 struct RequestInit;
 
 namespace cache {
@@ -381,6 +382,11 @@
 
   WebTaskScheduler* Scheduler();
   WebTaskScheduler* GetExistingScheduler() const;
+  void SetWebTaskSchedulingState(WebTaskSchedulingState* aState) override;
+
+  WebTaskSchedulingState* GetWebTaskSchedulingState() const override {
+    return mWebTaskSchedulingState;
+  }
 
   bool WindowInteractionAllowed() const;
 
@@ -417,6 +423,7 @@
   RefPtr<cache::CacheStorage> mCacheStorage;
   RefPtr<DebuggerNotificationManager> mDebuggerNotificationManager;
   RefPtr<WebTaskSchedulerWorker> mWebTaskScheduler;
+  RefPtr<WebTaskSchedulingState> mWebTaskSchedulingState;
   RefPtr<TrustedTypePolicyFactory> mTrustedTypePolicyFactory;
   uint32_t mWindowInteractionsAllowed = 0;
   bool mIsEligibleForMessaging{true};